rfd/0011-database-access.md

```diff
@@ -0,0 +1,698 @@
+sessions.
+
+### Session start
+
+`db.session.start` is emitted when a user has successfully connected to a database:
+
+```json
+{
+  "code": "TDB00I",
+  "service": "postgres",
+  "db_endpoint": "localhost:5432",
+  "db_protocol": "postgres",
+  "db_database": "postgres",
+  "db_user": "postgres",
+  "ei": 0,
+  "event": "db.session.start",
+  "namespace": "default",
+  "server_id": "5603cd43-1172-4f1f-8e35-2bf74dfecf15",
+  "sid": "427abfa3-fb1a-4a55-879c-77f5b6257cdb",
+  "time": "2020-11-06T03:50:20.802Z",
+  "uid": "f12b5199-5a1f-4e48-af4f-6980afffc48e",
+  "user": "dev"
+}
+```
+
+New fields are:
+
+* `service`: The name of Teleport database service handling the connection.
+* `db_endpoint`: The database instance address.
+* `db_protocol`: The database protocol.
+* `db_database`: The name of the database within DBMS a user is connecting to
+  for databases with protocol parsing support.
+* `db_user`: The name of the database user a user is connecting as for
+  databases with protocol parsing support.
+
+### Session end
+
+`db.session.end` is emitted when a user has disconnected from the database:
+
+```json
+{
+  "code": "TDB01I",
+  "service": "postgres",
+  "db_endpoint": "localhost:5432",
+  "db_protocol": "postgres",
+  "db_database": "test",
+  "db_user": "postgres",
+  "ei": 0,
+  "event": "db.session.end",
+  "sid": "56335587-0157-4228-9556-aaa416741ef7",
+  "time": "2020-11-06T03:50:20.802Z",
+  "uid": "49e5f882-40ea-4bdb-b462-4cccf3536ada",
+  "user": "dev"
+}
+```
+
+Same new fields as in the `db.session.start` event.
+
+### Query
+
+`db.query` is emitted when a user executes a database query:
+
+```json
+{
+  "code": "TDB02I",
+  "service": "postgres",
+  "db_endpoint": "localhost:5432",
+  "db_protocol": "postgres",
+  "db_database": "test",
+  "db_query": "SELECT 1;",
+  "db_user": "postgres",
+  "ei": 0,
+  "event": "db.query",
+  "sid": "a53b0e1c-42f0-43ad-bbdd-9f3e07b54c05",
+  "time": "2020-11-06T03:36:06.233Z",
+  "uid": "7af4ea5d-ba1a-42d0-96a6-0c41b9c89bb0",
+  "user": "dev"
+}
+```
+
+Same new fields as in the `db.session.start` event, plus:
+
+* `db_query`: Full text of the executed query for databases with protocol
+  parsing support.
+
+## CA rotation
+
+As explained above, a database server is configured with Teleport's host
+certificate authority and certificate/key pair signed by the host CA as well,
+produced by `tctl auth sign --type=db` command.
+
+This means that the database servers should be taken into consideration when
+performing Teleport's host CA rotation using `tctl auth rotate` command. See
+[Teleport documentation](https://gravitational.com/teleport/docs/admin-guide/#certificate-rotation)
+for more information about certificate rotation.
+
+Unlike Teleport nodes that automatically get reissued certificates signed by
+a new authority, we do not have control over database servers so a user must
+replace CA and keypair within the rotation grace period.
```

Comment:
`tctl auth rotate` should detect that DB access is configured and print instructions about rotating them

Reply:
Added a note below.